annotation-statistics/src/main/java/eu/europeana/annotation/statistics/service/AnnotationStatisticsService.java

```diff
@@ -0,0 +1,93 @@
+package eu.europeana.annotation.statistics.service;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.annotation.Resource;
+
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.json.BucketJsonFacet;
+
+import eu.europeana.annotation.definitions.model.vocabulary.AnnotationScenarioTypes;
+import eu.europeana.annotation.solr.exceptions.AnnotationServiceException;
+import eu.europeana.annotation.solr.service.SolrAnnotationService;
+import eu.europeana.annotation.solr.vocabulary.SolrAnnotationConstants;
+import eu.europeana.annotation.statistics.model.AnnotationMetric;
+import eu.europeana.annotation.statistics.model.AnnotationStatistics;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsClient;
+import eu.europeana.annotation.statistics.model.AnnotationStatisticsUser;
+import eu.europeana.annotation.statistics.vocabulary.AnnotationStatisticsConstants;
+
+public class AnnotationStatisticsService {
+
+	@Resource
+	SolrAnnotationService solrService;
+
+    public void getAnnotationsStatistics(AnnotationMetric annoMetric) throws AnnotationServiceException {
+    	annoMetric.setTimestamp(new Date());  
+    	//getting the annotations statistics for the clients
+    	QueryResponse annoFacetStatsJson = solrService.getAnnotationStatisticsJsonNestedFacets(SolrAnnotationConstants.GENERATOR_URI);
+    	extractAnnotationStatisticsClientUserFromJsonNestedFacet(annoFacetStatsJson, annoMetric, SolrAnnotationConstants.GENERATOR_URI, AnnotationStatisticsConstants.CLIENT);   	
+    	//getting the annotations statistics for the users
+    	annoFacetStatsJson = solrService.getAnnotationStatisticsJsonNestedFacets(SolrAnnotationConstants.CREATOR_URI);
+    	extractAnnotationStatisticsClientUserFromJsonNestedFacet(annoFacetStatsJson, annoMetric, SolrAnnotationConstants.CREATOR_URI, AnnotationStatisticsConstants.USER);   	
+    	//getting the overall annotations statistics
+    	annoFacetStatsJson = solrService.getAnnotationStatisticsJsonFacets(); 
+    	extractAnnotationStatisticsOverallFromJsonFacet(annoFacetStatsJson, annoMetric);
+    }
+    
+    private void extractAnnotationStatisticsClientUserFromJsonNestedFacet (QueryResponse annoFacetStats, AnnotationMetric annoMetric, String facetField, String target) {
+    	List<AnnotationStatistics> annotationStatistics = new ArrayList<AnnotationStatistics>();
+	    annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets();
+    	for (BucketJsonFacet mainBucket : annoFacetStats.getJsonFacetingResponse().getBucketBasedFacets(facetField).getBuckets()) {
+    		AnnotationStatistics annoStats = null;
+    		if(target.compareToIgnoreCase(AnnotationStatisticsConstants.CLIENT)==0) {
```

Comment:
please switch the position of the constant and use equalsIgnoreCase

Reply:
This code is not needed any more.